eyelid-match-ops/src/primitives/hamming.rs

```diff
@@ -0,0 +1,125 @@
+//! Implementation of the simple encoding
+
+use crate::primitives::poly::Poly;
+use ark_ff::Zero;
+use std::ops::{AddAssign, Sub};
+
+use rand::rngs::ThreadRng;
+
+use super::yashe::Yashe;
+use super::yashe::{Ciphertext, Message, PrivateKey, PublicKey, YasheConf};
+
+/// Contains the message to be encoded such that
+/// the Hamming distance can be computed later.
+pub struct SimpleHammingEncoding<C: YasheConf>
+where
+    C::Coeff: From<u128> + From<u64> + From<i64>,
+{
+    /// The message to be encoded
+    m: Message<C>,
+    /// The reverse of the message to be encoded
+    m_rev: Message<C>,
+}
+
+/// SimpleHammingEncodingCiphertext is a struct that holds two ciphertexts, c and c_rev,
+/// which are the encodings of the message m and m_rev, respectively. The encoding is
+/// done by reversing the message and encoding it as a regular Yashe Ciphertext.
+pub struct SimpleHammingEncodingCiphertext<C: YasheConf>
+where
+    C::Coeff: From<u128> + From<u64> + From<i64>,
+{
+    /// The ciphertext of the message m
+    c: Ciphertext<C>,
+    /// The ciphertext of the message m_rev
+    c_rev: Ciphertext<C>,
+}
+
+impl<C: YasheConf> SimpleHammingEncoding<C>
+where
+    C::Coeff: From<u128> + From<u64> + From<i64>,
+{
+    /// Creates a new `SimpleHammingEncoding` with the given message `m` and size `size`.
+    pub fn new(m: Message<C>, size: usize) -> Self {
+        let mut m_rev = Message {
+            m: Poly::<C>::zero(),
+        };
+        for i in 0..size {
+            m_rev.m[i] = m.m[size - i - 1];
+        }
+        Self { m, m_rev }
+    }
+
+    /// Sample a random SimpleHammingEncoding, by sampling a random binary Yashe Message, which
+    /// is done by calling function sample_binary_message, and returning a new SimpleHammingEncoding,
+    /// which sets m to the sampled message, and m_rev to the reverse of m.
+    pub fn sample(ctx: Yashe<C>, size: usize, rng: &mut ThreadRng) -> SimpleHammingEncoding<C> {
```

Comment:
Can this method be moved to a test-only module?

Here's an example of what that looks like for `Yashe`:
https://github.com/Inversed-Tech/eyelid/blob/ae61658ad1e5372e003cdf545f9cb2b181f4dff1/eyelid-match-ops/src/primitives/yashe.rs#L21-L22

https://github.com/Inversed-Tech/eyelid/blob/ae61658ad1e5372e003cdf545f9cb2b181f4dff1/eyelid-match-ops/src/primitives/yashe/test.rs#L20-L29